pkg/highlight/ftdetect.go

```diff
@@ -1,17 +1,33 @@
 package highlight
 
-import "regexp"
-
-// MatchFiletype will use the list of syntax definitions provided and the filename and first line of the file
-// to determine the filetype of the file
-// It will return the corresponding syntax definition for the filetype
-func MatchFiletype(ftdetect [2]*regexp.Regexp, filename string, firstLine []byte) bool {
-	if ftdetect[0] != nil && ftdetect[0].MatchString(filename) {
-		return true
+// MatchFileName will use the list of syntax definitions provided and the filename to
+// determine the filetype of the file
+func MatchFileName(header *Header, filename string) bool {
```

Comment:
Hmm, one more thought: we can do even nicer and make `MatchFileName()`, `HasFileSignature()` and `MatchFileSignature()` methods of `Header`.

Reply:
I like this idea and will prepare the change.